distributions/application/src/main/java/com/connexta/replication/application/Main.java

```diff
@@ -39,13 +46,29 @@ public static void main(String[] args) throws Exception {
     }
   }
 
-  @Bean
-  public ConfigFileReader configFileReader(
-      ReplicatorConfigManager configManager,
-      SiteManager siteManager,
-      ReplicatorHistoryManager historyManager) {
-    ConfigFileReader reader = new ConfigFileReader(configManager, siteManager, historyManager);
-    reader.run();
-    return reader;
+  public static void loadBootSystemProperties() {
+    String sslPropertiesPath = System.getProperty("ssl.system.properties");
+    if (sslPropertiesPath == null) {
+      sslPropertiesPath = "ssl.properties";
+    }
+    File sslPropertiesFile = new File(sslPropertiesPath);
+    if (!sslPropertiesFile.exists()) {
+      LOGGER.info("No ssl properties found at {}", sslPropertiesPath);
+      return;
+    }
+    Properties properties = new Properties();
+    try (FileInputStream fis = new FileInputStream(sslPropertiesFile)) {
+      properties.load(fis);
+      properties.forEach(
+          (name, value) -> {
```

Comment:
✏️ How about defining a setIfNotDefined() method that does the check and the set if not defined? 
This will simplify the call and make it more readable